### ISSUE TITLE
Recent change has broken Autocomplete for Autodesk's adsk module.

### ISSUE THREAD

**Issue by Zxynine** [microsoft/pylance-release#2261]
### VS Code version

1.63.2

### Extension version

For working:`v2021.12.1559732655` | For not working:`v2021.12.1559732655`

### OS type

Windows

### OS version

Windows_NT x64 10.0.19041 (windows 10)

### Python distribution

Other

### Python version

3.7.6

### Language server

Default

### Expected behaviour

I have a window ive left open for about a month now, so its settings have not synced. In a file from that window, if i start typing the name of a variable which is nested in a few modules under audtodesks adsk module, it gets suggested as per expected. See image:
![Working](https://user-images.githubusercontent.com/79198508/149634182-946e962d-2bea-4fb9-9158-1f859a951440.png)

### Actual behaviour

Within any new window, the autocomplete no longer shows for anything within the adsk module. It will still show if I access it with `adsk.core.matr...` but just starting to type `matr` will not show an option for `adsk.core.matrix3D`. See image:
![Failing](https://user-images.githubusercontent.com/79198508/149634272-33c5f34e-652c-44b3-82c2-a63889f2dc70.png)

### Steps to reproduce

I will be honest, I have no idea what I've done to cause this. I hope it was just an issue with a recent update, but it could be a problem from some extension I've installed or some changes made to the settings. Since nothing changed on my old windows that are open, I would not have seen the problem until I opened a new window.

### Logs

LOG FOR WORKING~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Experiment 'pythonaa' is active
Experiment 'pythonTensorboardExperiment' is active
Experiment 'PythonPyTorchProfiler' is active
Experiment 'pythonDeprecatePythonPath' is active
> conda info --json
> "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\python.exe" ~\.vscode\extensions\ms-python.python-2021.12.1559732655\pythonFiles\interpreterInfo.py
> D:\Tools\Python\python.exe ~\.vscode\extensions\ms-python.python-2021.12.1559732655\pythonFiles\interpreterInfo.py
> ~\.anaconda\Scripts\conda.exe info --json
> ~\.conda\Scripts\conda.exe info --json
> ~\.condarc\Scripts\conda.exe info --json
> conda --version
> ~\AppData\Local\Autodesk\webdeploy\production\4c84784f9b312b4c6ccb78d46529ebb70d7a5a09\Python\python.exe ~\.vscode\extensions\ms-python.python-2021.12.1559732655\pythonFiles\interpreterInfo.py
Python interpreter path: ~\AppData\Local\Autodesk\webdeploy\production\4c84784f9b312b4c6ccb78d46529ebb70d7a5a09\Python\python.exe
Starting Pylance language server.
[ERROR 2022-0-8 22:3:51.884]: Active interpreter type is detected as Unknown {"sysPrefix":"C:\\Users\\<User>\\AppData\\Local\\Autodesk\\webdeploy\\production\\4c84784f9b312b4c6ccb78d46529ebb70d7a5a09\\Python","envType":"Unknown","envName":"","envPath":"","path":"C:/Users/<User>/AppData/Local/Autodesk/webdeploy/production/4c84784f9b312b4c6ccb78d46529ebb70d7a5a09/Python/python.exe","architecture":3,"sysVersion":"3.7.6 (tags/v3.7.6:43364a7ae0, Dec 19 2019, 00:42:30) [MSC v.1916 64 bit (AMD64)]","version":{"raw":"3.7.6","major":3,"minor":7,"patch":6,"build":[],"prerelease":["final","0"]},"displayName":"Python 3.7.6 64-bit"}

```

LOG FOR NOT WORKING~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Experiment 'pythonaa' is active
Experiment 'pythonTensorboardExperiment' is active
Experiment 'PythonPyTorchProfiler' is active
Experiment 'pythonDeprecatePythonPath' is active
> conda info --json
> "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\python.exe" ~\.vscode\extensions\ms-python.python-2021.12.1559732655\pythonFiles\interpreterInfo.py
> D:\Tools\Python\python.exe ~\.vscode\extensions\ms-python.python-2021.12.1559732655\pythonFiles\interpreterInfo.py
> ~\AppData\Local\Autodesk\webdeploy\production\4c84784f9b312b4c6ccb78d46529ebb70d7a5a09\Python\python.exe ~\.vscode\extensions\ms-python.python-2021.12.1559732655\pythonFiles\interpreterInfo.py
Python interpreter path: ~\AppData\Local\Autodesk\webdeploy\production\4c84784f9b312b4c6ccb78d46529ebb70d7a5a09\Python\python.exe
> ~\.anaconda\Scripts\conda.exe info --json
> ~\.conda\Scripts\conda.exe info --json
> ~\.condarc\Scripts\conda.exe info --json
> conda --version
Starting Pylance language server.
[ERROR 2022-0-15 13:19:20.581]: Active interpreter type is detected as Unknown {"sysPrefix":"C:\\Users\\<User>\\AppData\\Local\\Autodesk\\webdeploy\\production\\4c84784f9b312b4c6ccb78d46529ebb70d7a5a09\\Python","envType":"Unknown","envName":"","envPath":"","path":"C:/Users/<User>/AppData/Local/Autodesk/webdeploy/production/4c84784f9b312b4c6ccb78d46529ebb70d7a5a09/Python/python.exe","architecture":3,"sysVersion":"3.7.6 (tags/v3.7.6:43364a7ae0, Dec 19 2019, 00:42:30) [MSC v.1916 64 bit (AMD64)]","version":{"raw":"3.7.6","major":3,"minor":7,"patch":6,"build":[],"prerelease":["final","0"]},"displayName":"Python 3.7.6 64-bit"}

```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by heejaechang** [comment]
the first window probably suggests the type since the type has been used somewhere else while vscode is running. but the second vscode didn't with indexing off.

it is happening since we currently only support the first level modules for third party libraries in indexing (hidden option)

but, now, "python.analysis.indexing" became a real option. set "python.analysis.indexing": true to enable indexing which will improve auto-import experience.

Also, new hidden option "python.analysis.packageIndexDepths" is added which can be used as below to control how much get indexed.

> "python.analysis.packageIndexDepths": [['sklearn', 2],['matplotlib', 2]],

By default, only the first level module of third party libraries will be indexed, but now, you can use ["package name", module levels to be scanned] to control that.

If you put "" (empty string) to package name, then it will affect all packages.

For your example, ["adsk", 3] will make up to 3 level sub modules to be picked up. 

Thank you

fix will be in the next release 2022.1.2

**Comment by Zxynine** [comment]
> but, now, "python.analysis.indexing" became a real option. set "python.analysis.indexing": true to enable indexing which will improve auto-import experience.

So, i searched in settings for it and did not see anything, so I attempted to directly add the option to my settings.json and it is saying "Unknown Configuration Setting". Everything should be up to date. Did I misunderstand what you were saying and that the option would be available in the next patch or something?

The same is true for ""python.analysis.packageIndexDepths": ".

How often do you guys release updates? I saw one was maybe like 5 days ago so depending on the frequency I may be waiting a bit.

**Comment by bschnurr** [comment]
release day is every wednesday, so should be in today.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.1.3, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202213-21-january-2022)
